diff --git a/src/docs.ts b/src/docs.ts
index 7c94593fa0cfc510b09541d8cdd7f9de0842c525..707424a70de6c96f33544b2680cceb620c506233 100644
--- a/src/docs.ts
+++ b/src/docs.ts
@@ -124,6 +124,8 @@ async function doProcessDocMetadata({
     lastUpdateFrontMatter,
   );
 
+  console.log("DOCS.TS: LASTUPDATE CONST: " + lastUpdate);
+
   // E.g. api/plugins/myDoc -> myDoc; myDoc -> myDoc
   const sourceFileNameWithoutExtension = path.basename(
     source,
@@ -176,7 +178,7 @@ async function doProcessDocMetadata({
   const title: string = frontMatter.title ?? contentTitle ?? baseID;
 
   const description: string = frontMatter.description ?? excerpt ?? '';
-
+  console.log("BITCH");
   const permalink = normalizeUrl([versionMetadata.path, docSlug]);
 
   function getDocEditUrl() {
@@ -209,6 +211,8 @@ async function doProcessDocMetadata({
   // Assign all of object properties during instantiation (if possible) for
   // NodeJS optimization.
   // Adding properties to object after instantiation will cause hidden
+
+  console.log("DOCS.TS: " + lastUpdate.lastUpdatedCommit);
   // class transitions.
   return {
     id,
@@ -225,6 +229,7 @@ async function doProcessDocMetadata({
     version: versionMetadata.versionName,
     lastUpdatedBy: lastUpdate.lastUpdatedBy,
     lastUpdatedAt: lastUpdate.lastUpdatedAt,
+    lastUpdatedCommit: lastUpdate.lastUpdatedCommit,
     sidebarPosition,
     frontMatter,
   };
